src/styles/utils/_variables.scss

```diff
@@ -0,0 +1,17 @@
+$color-day: #eee;
+$color-day-hover: rgb(226, 74, 74);
+$width-day: 100px;
+$font-size: 30px;
+$day-per-week: 7;
+$day-count: 29;
+$gap: 1px;
+$remaining-days: 31 - $day-count;
+$width-calendar: $day-per-week * $width-day + $gap * $day-per-week;
+$start-day: 
+  "mon" $width-calendar - (7 * $width-day) - 7px,
+  "tue" $width-calendar - (6 * $width-day) - 6px,
+  "wed" $width-calendar - (5 * $width-day) - 5px,
+  "thu" $width-calendar - (4 * $width-day) - 4px,
+  "fri" $width-calendar - (3 * $width-day) - 3px,
+  "sat" $width-calendar - (2 * $width-day) - 2px,
+  "sun" $width-calendar - (1 * $width-day) - 1px;

```

Comment:
don't repeat yourself: in these formulas you changed only 1-7 value, you can create cycle that will generate correct margin-left for needed selectors